cluster/watchdog.go

```diff
@@ -75,23 +80,76 @@ func (w *Watchdog) rescheduleContainers(e *Engine) {
 		// will abort because the name is already taken.
 		c.Engine.removeContainer(c)
 
-		newContainer, err := w.cluster.CreateContainer(c.Config, c.Info.Name, nil)
+		// keep track of all global networks this container is connected to
+		globalNetworks := make(map[string]*network.EndpointSettings)
+		// if the existing containter has global network endpoints,
+		// they need to be removed with force option
+		// "docker network disconnect -f network containername" only takes containername
```

Comment:
Confirmed this with @dongluochen. The only issue we noticed was conflicts in container name/id while searching, but that is a different unrelated issue.